### PR TITLE
feat: Use Debug Command Handler for Dalamud Commands

### PR DESCRIPTION
**Pull request by KazWolfe**
This pull request aims to improve the command handler experience by attaching a hook to the InvokeDebugCommand method. This allows Dalamud to receive all inbound commands, including those that have been directly injected or normally suppressed by `/merror off`. This replaces the legacy regex method of command evaluation, increasing speed and stability in case of system changes.

This method keeps Dalamud as the last command handler in the chain, so both standard and debug commands will take precedence. 

---

The InvokeDebugCommand method is called on any input to `ExecuteCommandInner`, which includes chat messages/non-commands. If a certain result is marked as a command but is not recognized, this method returns -1. If a certain result is a non-command (does not start with `/`), this method returns -2. InvokeDebugCommand is not called at all for "real" commands.

This hook will call `InvokeDebugCommand` normally and check its return code. If it matches `-1`, the hook passes the message contents to the Dalamud Command Handler for processing. If DCH succeeds in finding a command, this method will return `0` indicating to the game that a debug command was handled. If DCH fails to find a command, this method will return the original -1 causing the game to return the standard "command not found" response.

### CODE REVIEW

File: Dalamud/Game/Command/CommandManager.cs [hunk 1]
```diff
@@ -2,58 +2,52 @@
...
-        this.chatGui.CheckMessageHandled += this.OnCheckMessageHandled;
+        this.resolver = new CommandManagerAddressResolver();
+        this.resolver.Setup(scanner);
+        
+        this.testCommandHook = Hook<TestCommandDelegate>.FromAddress(
+            this.resolver.CommandErrorHandler,
+            this.OnTestCommand);
+        this.testCommandHook.Enable();
+
+        // this.chatGui.CheckMessageHandled += this.OnCheckMessageHandled;
```
Comment (goaaats) [inline Dalamud/Game/Command/CommandManager.cs:46]: What's going on here? Good to remove?

File: Dalamud/Game/Command/CommandManagerAddressResolver.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    /// <inheritdoc/>
+    protected override void Setup64Bit(ISigScanner sig)
+    {
+        this.SetupFramework(sig);
+    }
+    
+    private void SetupFramework(ISigScanner scanner)
+    {
+        this.CommandErrorHandler = 
+            scanner.ScanText("E8 ?? ?? ?? ?? 83 F8 ?? 74 ?? 83 F8 ?? 0F 85 ?? ?? ?? ?? 48 8B 07");
```
Comment (goaaats) [inline Dalamud/Game/Command/CommandManagerAddressResolver.cs:23]: Could we have this in CS?

**Review by goaaats — CHANGES_REQUESTED** [review]

File: Dalamud/Game/Command/CommandManagerAddressResolver.cs (hunk 2 above)
Comment (KazWolfe) [reply Dalamud/Game/Command/CommandManagerAddressResolver.cs:23]: https://github.com/aers/FFXIVClientStructs/pull/1102

**Comment by KazWolfe** [comment]
merge after 7.06hf2 stable release.

**Review by goaaats — APPROVED** [review]

**Pull request merged**